From 094abc4e39b9f12b4f833ff70754f43d505bb16e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E5=BF=98=E6=80=80?= <otgwt@outlook.com>
Date: Wed, 13 Nov 2024 18:42:14 +0800
Subject: [PATCH 232/360] net: phytmac_ethtool: Add struct
 kernel_ethtool_ts_info
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

2111375b85a: net: Add struct kernel_ethtool_ts_info
cause build error, so convert ethtool_ts_info to kernel_ethtool_ts_info

Log:
drivers/net/ethernet/phytium/phytmac_ethtool.c: In function ‘phytmac_get_ts_info’:
drivers/net/ethernet/phytium/phytmac_ethtool.c:204:45: error: passing argument 2 of ‘ethtool_op_get_ts_info’ from incompatible pointer type [-Werror=incompatible-pointer-types]
  204 |         return ethtool_op_get_ts_info(ndev, info);
      |                                             ^~~~
      |                                             |
      |                                             struct ethtool_ts_info *
In file included from drivers/net/ethernet/phytium/phytmac_ethtool.c:3:
./include/linux/ethtool.h:1205:59: note: expected ‘struct kernel_ethtool_ts_info *’ but argument is of type ‘struct ethtool_ts_info *’
 1205 |                            struct kernel_ethtool_ts_info *eti);
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
drivers/net/ethernet/phytium/phytmac_ethtool.c: At top level:
drivers/net/ethernet/phytium/phytmac_ethtool.c:522:43: error: initialization of ‘int (*)(struct net_device *, struct kernel_ethtool_ts_info *)’ from incompatible pointer type ‘int (*)(struct net_device *, struct ethtool_ts_info *)’ [-Werror=incompatible-pointer-types]
  522 |         .get_ts_info                    = phytmac_get_ts_info,
      |                                           ^~~~~~~~~~~~~~~~~~~
drivers/net/ethernet/phytium/phytmac_ethtool.c:522:43: note: (near initialization for ‘phytmac_ethtool_ops.get_ts_info’)
cc1: all warnings being treated as errors
---
 drivers/net/ethernet/phytium/phytmac_ethtool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/phytium/phytmac_ethtool.c b/drivers/net/ethernet/phytium/phytmac_ethtool.c
index 592d2d9dc6d4..d212451cd682 100644
--- a/drivers/net/ethernet/phytium/phytmac_ethtool.c
+++ b/drivers/net/ethernet/phytium/phytmac_ethtool.c
@@ -176,7 +176,7 @@ static int phytmac_set_ringparam(struct net_device *ndev,
 }
 
 static int phytmac_get_ts_info(struct net_device *ndev,
-			       struct ethtool_ts_info *info)
+			       struct kernel_ethtool_ts_info *info)
 {
 	struct phytmac *pdata = netdev_priv(ndev);
 
-- 
2.20.1

